File: ApplicationAlgorithm/test/cpp/unit1.cpp
#include <iostream>

using namespace std;

int fibonacci(int n)
{
	if (n < 2)
		return n;
	return fibonacci(n - 1) + fibonacci(n - 2);
}

int function(int k)
{
	if (k < 1)
		return 0;
	int i = 0;
	while (true)
	{
		if (fibonacci(i) > k)
			return fibonacci(i);
		i++;
	}
}

int main()
{
	int k;
	cin >> k;
	cout << function(k);
}

File: ApplicationAlgorithm/test/py/unit1.py
def fibonacci(n):
    if n < 2:
        return n
    else:
        return fibonacci(n - 1) + fibonacci(n - 2)

def function(k):
    if k < 1:
        return 0
    else:
        i = 0
        n = 0
        while (n <= k):
            n = fibonacci(i)
            i += 1
        return n

k = int(input())
print(function(k))